csrc/host_ir/jit.cpp

```diff
@@ -0,0 +1,362 @@
+          : allocate->buffer()->dtype());
+  llvm::Value* dtype_constant =
+      llvm::ConstantInt::get(int32_type, static_cast<int32_t>(data_type));
+  llvm::Value* device_index_constant =
+      llvm::ConstantInt::get(int64_type, host_ir_jit_params.getCommunicator()->deviceId());
+
+  // Get the at::native::empty_strided_cuda function pointer (registered in the
+  // JIT)
+  llvm::Function* at_empty_strided_cuda_func =
+      mod->getFunction("at::native::empty_strided_cuda");
+  if (at_empty_strided_cuda_func == nullptr) {
+    llvm::FunctionType* at_empty_strided_cuda_func_type =
+        llvm::FunctionType::get(
+            builder.getVoidTy(),
+            {int64_ptr_type,
+             int64_type,
+             int64_ptr_type,
+             int64_type,
+             int32_type,
+             int64_type,
+             void_ptr_type},
+            false);
+    at_empty_strided_cuda_func = llvm::Function::Create(
+        at_empty_strided_cuda_func_type,
+        llvm::Function::ExternalLinkage,
+        "at::native::empty_strided_cuda",
+        mod);
+  }
+
+  // Call at::native::empty_strided_cuda with constants
+  builder.CreateCall(
+      at_empty_strided_cuda_func,
+      {sizes_arg,
+       ndim_arg,
+       strides_arg,
+       strides_ndim_arg,
+       dtype_constant,
+       device_index_constant,
+       out_tensor_arg});
+
+  // Return the result
+  builder.CreateRetVoid();
+
+  // Verify the module
+  std::string error;
+  llvm::raw_string_ostream error_stream(error);
+  NVF_ERROR(!llvm::verifyModule(*mod, &error_stream), "LLVM module verification failed: " + error);
+
+  const bool debug_print = isDebugDumpEnabled(DebugDumpOption::HostIrJit);
+  if (debug_print) {
+    // Print the LLVM IR module
+    llvm::outs() << "=== LLVM IR ===\n";
+    mod->print(llvm::outs(), nullptr);
+  }
+}
+
+void compile(const hir::HostIrContainer* container, llvm::orc::LLJIT* jit, std::unordered_map<const kir::Allocate*, allocate_fn>& allocate_funcs_, const HostIrJitParams& host_ir_jit_params) {
+  FUSER_PERF_SCOPE("HostIrJit::compile");
+  if (allocate_funcs_.size() > 0) {
+    return;
+  }
+  
+  if (container == nullptr) {
+    NVF_ERROR(false, "container is nullptr during host ir JIT compilation");
+    return;
+  }
+  auto ctx = std::make_unique<llvm::LLVMContext>();
+  auto mod = std::make_unique<llvm::Module>(
+      "host_ir_container_" +
+          std::to_string(reinterpret_cast<uintptr_t>(container)),
+      *ctx);
+  
+  std::unordered_map<const kir::Allocate*, std::string> allocate_func_names;
+  for (auto expr : container->topLevelExprs()) {
+    if (auto allocate = dynamic_cast<const kir::Allocate*>(expr)) {
+      // Generate a unique function name for this allocate
+      generateAllocateFunc(allocate, mod.get(), host_ir_jit_params);
+      // Store the mapping from allocate to function name
+      allocate_func_names[allocate] = hostIrJitAllocateFuncName + "_" +
+          std::to_string(reinterpret_cast<uintptr_t>(allocate));
```

Comment:
Can `allocate->name()` or `ir_utils::varName(buffer)` be used as unique IDs? They are more readable than pointer addresses.